drivers/arm/ws2812_pwm.c

```diff
@@ -1 +1,207 @@
-#error("NOT SUPPORTED")
\ No newline at end of file
+#include "ws2812.h"
+#include "quantum.h"
+#include "hal.h"
+
+/* Adapted from https://github.com/joewa/WS2812-LED-Driver_ChibiOS/ */
+
+#ifdef RGBW
+#    error "RGBW not supported"
+#endif
+
+#ifndef WS2812_PWM_DRIVER
+#    define WS2812_PWM_DRIVER PWMD2  // TIMx
+#endif
+#ifndef WS2812_PWM_CHANNEL
+#    define WS2812_PWM_CHANNEL 2  // Channel
+#endif
+#ifndef WS2812_PWM_PAL_MODE
+#    define WS2812_PWM_PAL_MODE 2  // DI Pin's alternate function value
+#endif
+#ifndef WS2812_DMA_STREAM
+#    define WS2812_DMA_STREAM STM32_DMA1_STREAM2  // DMA Stream for TIMx_UP
+#endif
+#ifndef WS2812_DMA_CHANNEL
+#    define WS2812_DMA_CHANNEL 2  // DMA Channel for TIMx_UP
+#endif
+
+#ifndef WS2812_PWM_TARGET_PERIOD
+//#    define WS2812_PWM_TARGET_PERIOD 800000 // Original code is 800k...?
+#    define WS2812_PWM_TARGET_PERIOD 80000  // TODO: work out why 10x less on f303/f4x1
+#endif
+
+/* --- PRIVATE CONSTANTS ---------------------------------------------------- */
+
+#define WS2812_PWM_FREQUENCY (STM32_SYSCLK / 2)                             /**< Clock frequency of PWM, must be valid with respect to system clock! */
```

Comment:
The timer's frequency depends on the specific APB1 or APB2's frequency, and which was decided by the APB divider defined on the mcuconf.h. If anyone tuned the value, the PWM frequency will be incorrect, so it better to make this MACRO overridable as the pwm driver and time channel selections.

Reply:
Feel free to add an incremental PR, but as this has been merged its too late to change it within this PR.